modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -615,194 +646,246 @@ GDScriptParser::DataType GDScriptAnalyzer::resolve_datatype(GDScriptParser::Type
 	return result;
 }
 
-void GDScriptAnalyzer::resolve_class_interface(GDScriptParser::ClassNode *p_class) {
-	if (p_class->resolved_interface) {
+void GDScriptAnalyzer::resolve_class_member(GDScriptParser::ClassNode *p_class, GDScriptParser::ClassNode::Member &p_member) {
+	if (p_member.get_datatype().is_resolving()) {
+		push_error(vformat(R"(Could not resolve member "%s": Cyclic reference.)", p_member.get_name()), p_member.get_source_node());
+		return;
+	}
+
+	if (p_member.get_datatype().is_set()) {
 		return;
 	}
-	p_class->resolved_interface = true;
+
+	GDScriptParser::DataType base_type = p_class->base_type;
+	if (base_type.kind == GDScriptParser::DataType::CLASS) {
+		GDScriptParser::ClassNode *base_class = base_type.class_type;
+		if (parser->has_class(base_class)) {
+			resolve_class_interface(base_class);
+		} else {
+			Ref<GDScriptParserRef> parser_ref = get_parser_for(base_type.script_path);
+			if (parser_ref.is_null()) {
+				push_error(vformat(R"(Could not find script "%s".)", base_type.script_path), p_class);
+			} else {
+				Error err = parser_ref->raise_status(GDScriptParserRef::INTERFACE_SOLVED);
+				if (err) {
+					push_error(vformat(R"(Could not resolve base class "%s": %s.)", base_class->fqcn, error_names[err]), p_class);
+				}
+				if (!base_class->resolved_interface) {
+					if (!parser_ref->get_parser()->has_class(base_class)) {
+						ERR_PRINT(R"(Parser bug: Mismatched external parser.)");
+					}
+					push_error(vformat(R"(Could not resolve base class "%s": Cyclic reference.)", base_class->fqcn), p_class);
+				}
+			}
+		}
+	}
 
 	GDScriptParser::ClassNode *previous_class = parser->current_class;
 	parser->current_class = p_class;
 
-	for (int i = 0; i < p_class->members.size(); i++) {
-		GDScriptParser::ClassNode::Member member = p_class->members[i];
-
-		switch (member.type) {
+	{
```

Comment:
this block is just here to make the diff is more readable

Reply:
Oh, I understand now what you meant. No need for a block, as in the settings of Github, you can hide whitespace and change the way Github shows the diff.

![image](https://user-images.githubusercontent.com/270928/205414845-0a1771cd-d32c-429a-92a5-2b9042d8f7f3.png)

Reply:
ooo ok

Reply:
this actually doesn't make it more readable for whatever reason